packages/synapse-react-client/src/components/Forum/DiscussionThread.tsx

```diff
@@ -49,6 +50,7 @@ export function DiscussionThread(props: DiscussionThreadProps) {
   const [showSignInModal, setShowSignInModal] = useState(false)
   const [showRestoreModal, setShowRestoreModal] = useState(false)
   const [showSubscriberModal, setShowSubscriberModal] = useState(false)
+  const [replies, setReplies] = useState<DiscussionReplyBundle[]>([])
```

Comment:
Make the presence of the search param the source of truth for this 'state' and compute everything instead of using `useState` + `useEffect`

```suggestion
```

Suggestion continued below